File: sample-legacy/src/main/java/com/example/legacy/TrickyMath.java
package com.example.legacy;

public class TrickyMath {

    public int safeAdd(int a, int b) {
        long sum = (long) a + (long) b;
        if (sum > Integer.MAX_VALUE) {
            return Integer.MAX_VALUE;
        } else if (sum < Integer.MIN_VALUE) {
            return Integer.MIN_VALUE;
        }
        return (int) sum;
    }

    public String formatCurrency(double amount) {
        if (Double.isNaN(amount) || Double.isInfinite(amount)) {
            return "INVALID";
        }
        long cents = Math.round(amount * 100.0);
        long dollars = cents / 100;
        long remCents = Math.abs(cents % 100);
        return String.format("$%d.%02d", dollars, remCents);
    }

    public boolean isBitSet(int number, int bitIndex) {
        if (bitIndex < 0 || bitIndex >= 32) {
            return false;
        }
        return (number & (1 << bitIndex)) != 0;
    }
}

File: archived-runs/run-2-simpletodo-task-success/migrated-src/com/example/legacy/TrickyMath.kt
package com.example.legacy

class TrickyMath {

    fun safeAdd(a: Int, b: Int): Int {
        val sum = a.toLong() + b.toLong()
        return when {
            sum > Integer.MAX_VALUE -> Integer.MAX_VALUE
            sum < Integer.MIN_VALUE -> Integer.MIN_VALUE
            else -> sum.toInt()
        }
    }

    fun formatCurrency(amount: Double): String {
        if (amount.isNaN() || amount.isInfinite()) {
            return "INVALID"
        }
        val cents = Math.round(amount * 100.0)
        val dollars = cents / 100
        val remCents = Math.abs(cents % 100)
        return String.format("$%d.%02d", dollars, remCents)
    }

    fun isBitSet(number: Int, bitIndex: Int): Boolean {
        if (bitIndex < 0 || bitIndex >= 32) {
            return false
        }
        return number and (1 shl bitIndex) != 0
    }
}